cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -851,33 +852,24 @@ class lists_column_wrapper : public detail::column_wrapper {
     auto valids = cudf::test::make_counting_transform_iterator(
       0, [&v](auto i) { return v.empty() ? true : v[i]; });
 
-    // preprocess the incoming lists. unwrap any "root" lists and just use their
-    // underlying non-list data.
-    // also, sanity check everything to make sure the types of all the columns are the same
+    // compute the expected hierarchy
+    column_view expected_hierarchy;
+    int expected_depth = -1;
+    std::for_each(elements.begin(),
+                  elements.end(),
+                  [&expected_depth, &expected_hierarchy](lists_column_wrapper const& lcw) {
+                    if (lcw.depth > expected_depth) {
+                      expected_depth     = lcw.depth;
+                      expected_hierarchy = lcw.get_view();
+                    }
+                  });
```

Comment:
You can use a `std::accumulate` with ` std::pair` here. 
```suggestion
    // compute the expected hierarchy - user C++17 structures bindings when available
    auto const hierarchy_and_depth = std::accumulate(
      elements.begin(),
      elements.end(),
      std::pair<column_view, int32_t>{{}, -1},
      [](auto acc, lists_column_wrapper const& lcw) {
        return lcw.depth > acc.second ? std::make_pair(lcw.get_view(), lcw.depth) : acc;
      });

    auto expected_hierarchy   = hierarchy_and_depth.first;
    auto const expected_depth = hierarchy_and_depth.second;
```